connector/connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala

```diff
@@ -148,29 +147,25 @@ class SparkConnectProtoSuite extends PlanTest with SparkConnectPlanTest {
   }
 
   test("Aggregate with more than 1 grouping expressions") {
-    withSQLConf(SQLConf.DATAFRAME_RETAIN_GROUP_COLUMNS.key -> "false") {
-      val connectPlan =
-        connectTestRelation.groupBy("id".protoAttr, "name".protoAttr)()
-      val sparkPlan =
-        sparkTestRelation.groupBy(Column("id"), Column("name")).agg(Map.empty[String, String])
-      comparePlans(connectPlan, sparkPlan)
-    }
+    val connectPlan =
+      connectTestRelation.groupBy("id".protoAttr, "name".protoAttr)()
+    val sparkPlan =
+      sparkTestRelation.groupBy(Column("id"), Column("name")).agg(Map.empty[String, String])
+    comparePlans(connectPlan, sparkPlan)
   }
 
   test("Aggregate expressions") {
-    withSQLConf(SQLConf.DATAFRAME_RETAIN_GROUP_COLUMNS.key -> "false") {
```

Comment:
`DATAFRAME_RETAIN_GROUP_COLUMNS` will take no effect in Connect, shoud we mention it somewhere?

Reply:
There are two questions:
1. Do we want to support both? If not, maybe go with a documentation?
2. if we support both ways, by what approach? Should that be a part of proto (stateless) or we should support client side config?

I am trying to get these questions discussed a bit this week to close this chapter.

Reply:
> we should support client side config?

Reply:
makes sense. Let me drive a consensus on this. The usage `__repr__` probably can vote +1 on the client side configuration.

Reply:
There are couple of things to point out:

1) The behavior can be controlled by applying the default values for the execution on the server side and I would like to hold adding knobs until we have real evidence that we need it.

2) `__repr__` the behavior of REPL is defined by PySpark and we don't have to follow it to the letter. We can make independent decisions that define what the behavior on `__repr__` is. For example, I don't think we should eagerly evaluate.